packages/fetchai/skills/generic_seller/generic_data_model.py

```diff
@@ -18,20 +18,36 @@
 # ------------------------------------------------------------------------------
 
 """This package contains the dataModel for the generic seller aea."""
+import logging
+import sys
+from typing import Any, Dict
 
 from aea.helpers.search.models import Attribute, DataModel
 
-SCHEME = {"country": "UK", "city": "Cambridge"}
+SUPPORTED_TYPES = {"str": str, "int": int, "float": float, "bool": bool}
+
+logger = logging.getLogger(__name__)
 
 
 class Generic_Data_Model(DataModel):
     """Data model for the generic seller aea."""
 
-    def __init__(self):
+    def __init__(self, data_model_dict: Dict[str, Any]):
         """Initialise the dataModel."""
-        self.attribute_country = Attribute("country", str, True)
-        self.attribute_city = Attribute("city", str, True)
+        self.attributes = []
+        try:
```

Comment:
I would add assert statements to ensure that the types are correct for the dictionary values